app/lib/view/moodle.dart

```diff
@@ -104,237 +103,347 @@ class _MoodleWebViewState extends State<MoodleWebView> {
+                              uri.rawValue.contains(
+                                  ".schulportal.hessen.de/index.php?logout=all")) {
+                            return NavigationActionPolicy.CANCEL;
+                          }
+
+                          if (!uri.rawValue
+                              .contains(".schulportal.hessen.de")) {
+                            await launchUrl(uri);
+
+                            return NavigationActionPolicy.CANCEL;
+                          }
+
+                          return NavigationActionPolicy.ALLOW;
+                        },
+                        onLoadStart: (controller, uri) async {
+                          if (await controller.canGoBack()) {
+                            canGoBack.value = true;
+                          } else {
+                            canGoBack.value = false;
+                          }
+
+                          if (await controller.canGoForward()) {
+                            canGoForward.value = true;
+                          } else {
+                            canGoForward.value = false;
+                          }
+                        },
+                        onLoadStop: (controller, url) async {
+                          pullToRefreshController!.endRefreshing();
+                          progressIndicator.value = 0;
+                        },
+                        onPageCommitVisible: (controller, uri) async {
+                          if (!loggedIn.value) {
+                            if (!reachedLogin && uri!.rawValue.contains("singleSignOn") && !uri.rawValue.contains(client.schoolID)) {
+                              controller.loadUrl(
+                                  urlRequest: URLRequest(
+                                      url: WebUri("${uri.rawValue}&i=${client.schoolID}")
+                                  )
+                              );
+                              reachedLogin = true;
                             }
 
-                            final canGoBack = await webViewController!.canGoBack();
-                            if (canGoBack) {
-                              webViewController!.goBack();
-                            } else {
-                              hideWebView.value = true;
-                              Navigator.pop(context);
+                            if (reachedLogin) {
+                              controller.evaluateJavascript(
+                                  source:
+                                  """
+                                  setTimeout(function() {
+                                    document.querySelector('#username2').value = '${client.username}';
+                                  }, 250);
+                                  
+                                  setTimeout(function() {
+                                    document.querySelector('#inputPassword').value = '${client.password}';
```

Comment:
This would cause an SQL-Injection like error if the special character ' is used in the query. It should be prefixed with an \ to tell the JS engine to take it as string.